sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1252,7 +1252,9 @@ class Analyzer(
           case ae: AnalysisException => filter
         }
 
-      case sort @ Sort(sortOrder, global, aggregate: Aggregate) if aggregate.resolved =>
+      case sort @ Sort(sortOrder, global, aggregate: Aggregate)
+        if aggregate.resolved && sortOrder.forall(x => IntegerIndex.unapply(x.child).isEmpty) =>
```

Comment:
We have a conf `conf.orderByOrdinal` to control whether the integer values are parsed as position. Thus, the current fix ignores this conf. Could you fix it? Also added a test case to ensure both options are covered. That is, `true` and `false`